### DISCUSSION TITLE
Skip Step if element is not on a page

### DISCUSSION THREAD

**Discussion by tyranthacker** [shipshapecode/shepherd#2304] [Q&A]
There are some instances were an element won't appear on the page for certain types of users. I want to skip a step if the element does not exist. I have red the issues here on github but can't seem to find a straight answer or an example on how to do this properly.

**Comment by RobbieTheWagner** [comment]
@tyranthacker if the element won't exist, you can omit it from your tour. I would create different tours for your different types of users.

**Reply by tyranthacker** [reply]
That’s not possible for this situation but I thought there was a way to skip a step if the element doesn’t exist:

action: A function executed when the button is clicked on. It is automatically bound to the tour the step is associated with, so things like this.next will work inside the action. You can use action to skip steps or navigate to specific steps..

**Reply by tyranthacker** [reply]
Or

showOn: A function that, when it returns true, will show the step. If it returns false, the step will be skipped.

**Reply by RobbieTheWagner** [reply]
Sure, I would try those then, but again, if you know an element won't exist, don't add it to your tour.

**Reply by tyranthacker** [reply]
I am working on a fix. Some elements will exist sometimes and not other items and silly to write the same tour but slightly different multiple times because I can't skip a step, still don't understand why there isn't a skip step. I appreciate your help!

**Reply by tyranthacker** [reply]
I'll read this, I didnt see this before I posted. I saw a different post. #1437

**Reply by RobbieTheWagner** [reply]
@tyranthacker you mentioned two options that could work above here. `showOn` sounds like it would work for you. I would still recommend just not including the step in the tour, if you know you don't need it. You don't have to duplicate the tours, just programmatically build the `steps`.
